### PR TITLE
MYNEWT-789 Add RuuviTag polling LIS2DH12 3-axis accel SPI driver

### PR DESCRIPTION
**Pull request by vrahane**
- Adding basic SPI driver for LIS2DH12 to work with the Ruuvitag
- Changing the frequency for SPI to 4Mhz.
- Change Ruuvi sample rate for LIS2DH12

### CODE REVIEW

File: hw/drivers/sensors/bme280/src/bme280.c [hunk 1]
```diff
@@ -53,7 +53,7 @@ double NEG_INF = -1.0/0.0;
 static struct hal_spi_settings spi_bme280_settings = {
     .data_order = HAL_SPI_MSB_FIRST,
     .data_mode  = HAL_SPI_MODE0,
-    .baudrate   = 500,
+    .baudrate   = 4000,
```
Comment (mkiiskila) [inline hw/drivers/sensors/bme280/src/bme280.c]: Did you mean to bump the SPI datarate for this other sensor too?

File: hw/drivers/sensors/lis2dh12/src/lis2dh12.c [hunk 2]
```diff
@@ -0,0 +1,866 @@
...
+ */
+
+#include <assert.h>
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+
+#include "defs/error.h"
+#include "os/os.h"
+#include "sysinit/sysinit.h"
```
Comment (mkiiskila) [inline hw/drivers/sensors/lis2dh12/src/lis2dh12.c]: Maybe include sysinit before os.h? I can't remember if there were any MYNEWT_VAL() checks in those header files, but it is quite possible.

File: hw/drivers/sensors/bme280/src/bme280.c (hunk 1 above)
Comment (vrahane) [inline hw/drivers/sensors/bme280/src/bme280.c]: Yes, it was intentional. I tested it as well.

File: hw/drivers/sensors/lis2dh12/src/lis2dh12.c (hunk 2 above)
Comment (vrahane) [inline hw/drivers/sensors/lis2dh12/src/lis2dh12.c]: done

**Pull request merged**